### ISSUE TITLE
why use explicit weight decay in training

### ISSUE THREAD

**Issue by zhoulikeda** [spindro/AP-GCN#2]
The l2_reg is computed through parameters in the model itself, then why not use optimizer = Adam(model.parameters(),lr=lr, weight_decay=weight_decay) in pytorch?

**Comment by spindro** [comment]
Because it's applied just on the first layer.
Actually, newer version of pytorch should let you that by specifying a list
of parameters instead of the entire model when you initialize adam.

**Comment by zhoulikeda** [comment]
OK，thanks for your reply. Then I wondered why weight_decay is applied just on the first layer instead of all layers？











At 2020-03-13 16:14:32, "Indro Spinelli" <notifications@github.com> wrote:

Because it's applied just on the first layer.
Actually, newer version of pytorch should let you that by specifying a list
of parameters instead of the entire model when you initialize adam.

**Comment by spindro** [comment]
As you may know, weight_decay is a regularization term. Similarly to APPNP,
we used a fully connected network with just two layers.  To apply the
regularization on both layers is too much
but is way more effective to apply just on the first layer which in this
case has the "highest burden".

**Comment by zhoulikeda** [comment]
Thanks a lot.
